### DISCUSSION TITLE
Web Analytics Export

### DISCUSSION THREAD

**Discussion by RyanGaudion** [vercel/community#2587] [Idea]
Currently Web Analytics are only available for:
- Hobby - 1 month
- Pro - 1 Year
- Add-on - 2 Years

However, I would love to be able to hold this data forever. Of course, I can't expect Vercel to host it forever, so having an export ability for Web Analytics would be great.

It means that my usage data is never lost (so I can compare year on year) while also not requiring Vercel to host forever.

A simple csv/json export would be great for the main stats:

- visitors per day (+top pages & top referrers)
- page views per day (+top pages & top referrers

**Comment by z4nr34l** [comment]
I would love to see and use this data in axiom - which is currently used only for log drains.

**Comment by shyd0w** [comment]
any updates for this?

**Comment by moinulmoin** [comment]
any update on this? I also want to export my analytics data

**Comment by g1sbi** [comment]
Bump, would love this feature

**Comment by Figumari** [comment]
I would also be very interested in such a feature. 🎁

**Comment by james-beston** [comment]
Agreed… any news on this would be great! I love the ease of use of VA, but I could make so much more use of this if I could add the data to a staff dashboard.

**Comment by bwat-dev** [comment]
I need this functionality too

**Comment by geofot96** [comment]
Same here

**Comment by JayeshVP24** [comment]
We need this Vercel

**Comment by EricAndresen** [comment]
This feature makes the difference between being able to continue to use Vercel Analytics as we scale and needing to migrate to a third party solution

**Comment by craigRSA** [comment]
Have a look at this post
https://github.com/orgs/vercel/discussions/1068#discussioncomment-7403107

**Comment by andreeasteriu** [comment]
any updates?

**Comment by amyegan** [comment]
Hey everyone! You can export Analytics data as a CSV now 🎉

https://vercel.com/changelog/csv-export-in-web-analytics

**Reply by jordan-smith-et** [reply]
Export granular data of all interactions so we can do things like "how many people clicked this button on that day" which VA does not have currently. That's what I need

**Reply by amyegan** [reply]
Hey @jordan-smith-et! The original question asked about a CSV/JSON export, which is available now.

> A simple csv/json export would be great for the main stats:
> 
> visitors per day (+top pages & top referrers)
> page views per day (+top pages & top referrers

It sounds like you're looking for more granular data user interactions on each page. You can [track custom events](https://vercel.com/docs/analytics/custom-events#tracking-custom-events) with Vercel Analytics on Pro and Enterprise plans. That would let you track user interactions, like button clicks. [Log Drains](https://vercel.com/docs/observability/logs#log-drains) and [OTEL](https://vercel.com/docs/observability/otel-overview#quickstart-for-using-the-vercel-opentelemetry-collector) are options as well.

**Reply by jordan-smith-et** [reply]
Corrected. 

However, my point remains. You can't download a .csv with each row corresponding to an interaction, as far as I can see? you just get aggregated data, which isn't helpful for my use case "how many people did [event] on [day]"

[a comment by an automated account is not shown]
